File: 프로그래머스/JS/진료순서 정하기.js
/* 
첫번째, sorted를 해서 리스트업(내림차순)을 하기
두번째, 리스트업된 리스트의 인덱스를 기준으로 replace 함수 쓰는 건 어떨까?

참고_1 : 배열의 각 요소에 대해 지정된 함수를 실행하고, 그 결과를 모아 새로운 배열로 반환
*/

function solution(emergency) {
    const sortedEmergency = [...emergency].sort((a, b) => b - a);
    
    const answer = emergency.map(value => {
        // 3. 각 원본 값이 내림차순 정렬 배열에서 몇 번째인지 인덱스 검색 후 1 더하기
        return sortedEmergency.indexOf(value) + 1;
    });

    return answer;
}

File: 프로그래머스/파이썬/진료순서 정하기.py
''' 
첫번째, sorted를 해서 리스트업(내림차순)을 하기
두번째, for 반복문을 활용, 리스트업된 리스트(=리스트업)의 기준으로 순차적으로 값을 도출하고
세번째, emergency를 딕셔너리로 변용 후 활용해서, 값을 기준으로 mapping을 한다.
네번째, emergency의 값을 각각 리스트업의 '인덱스 + 1' 값들로 변환, 이때는 replace 함수 활용
'''

def solution(emergency):
    sorted_emergency = sorted(emergency, reverse=True) # 원래 기본이 내림차순
    answer = []
    
    for e in emergency:
        rank = sorted_emergency.index(e) + 1  # 인덱스는 0부터 시작하므로 +1
        answer.append(rank)
        
    return answer